app/Http/Controllers/MovieController.php

```diff
@@ -18,22 +18,26 @@ public function index(Request $request)
     {  
         $searchBy = '';
         $searchQuerey ='';
+        $searchByGenre ='';
+        $searchQuereyGenre ='';
         if ($request->has('title')){
             $searchQuerey = $request->query('title'); 
             $searchBy = 'title';
-        } else if ($request->has('description')){
+        } if ($request->has('description')){
             $searchQuerey = $request->query('description');
             $searchBy ='description';
-        }else if($request->has('url')){
+        } if($request->has('url')){
             $searchQuerey = $request->query('url');
             $searchBy = 'url';
-        }else if($request->has('genre')){
-            $searchQuerey = $request->query('genre');
-            $searchBy = 'genre';
+        }if($request->has('genre')){
+            $searchQuereyGenre = $request->query('genre');
+            $searchByGenre = 'genre';
```

Comment:
Why if instead of else if? If everything is if, your variable can change multiple times and it will be unpredictable, you need to change the logic if you want to support multiple things. This way you are stepping over your own code and only one of those if's will be valid, and that is, the last one that is true